lpc845-test-stand/test-assistant/src/handle_msg.rs

```diff
@@ -329,50 +359,77 @@ use crate::{PININT0_PIN, PIO0_8};
 /// - we don't try to change our accidentally dynamic pins (RTS/CTS)
 fn handle_set_direction_dynamic(
     pin: DynamicPin,
-    direction: pin::Direction,
-    // TODO LSS move this conversion to centrl place. should this really happen here?
-    level: crate::pin::Level,
+    direction: ContextualDirection,
     pinint0_pin: &mut GpioPin<PININT0_PIN, Dynamic>,
-    // TODO(LSS) remove this once cts is single direction again
-    cts: &mut GpioPin<PIO0_8, Dynamic>,
     dynamic_int_pin_levels: &mut FnvIndexMap<usize, (pin::Level, Option<u32>), U4>,
     dyn_noint_pins: &mut crate::resources::dyn_noint_pins,
-) -> Result<(), PinReadError> {
-    // TODO(LSS) move input switch here as well
-
-    // TODO move this level conversion to oen central place
-    let gpio_level: gpio::Level = level.into();
+) -> Result<(), DynamicPinError> {
+    // Check if this pin has a mapped pin number (e.g. is it
+    // allocated as a GPIO)
+    let pin_number = if let Some(pn) = pin.get_pin_number() {
```

Comment:
This could be shortened to `pin.get_pin_number().ok_or_else(|| DynamicPinError::NotGpioPin)?;`